contracts/tokenbridge/ethereum/L1AtomicTokenBridgeCreator.sol

```diff
@@ -662,24 +662,23 @@ contract L1AtomicTokenBridgeCreator is Initializable, OwnableUpgradeable {
     }
 
     /**
-     * @notice Generate a creation code that results on a contract with `code` as bytecode.
-     *         Source - https://github.com/0xsequence/sstore2/blob/master/contracts/utils/Bytecode.sol
-     * @param code The returning value of the resulting `creationCode`
-     * @return creationCode (constructor) for new contract
+     * @notice Generate a creation code that results with a contract with `code` as deployed code.
+     *         Generated creation code shall match the one generated by Solidity compiler.
+     * @dev Prepended constructor bytecode consists of:
+     *      - 608060405234801561001057600080fd5b50 - store free memory pointer, then check no callvalue is provided
+     *      - 61xxxx - push 2 bytes of `code` length
+     *      - 806100206000396000f3fe - copy deployed code to memory and return the location of it
+     * @param code Deployed bytecode to which constructor bytecode will be prepended
+     * @return Creation code of a new contract
      */
     function _creationCodeFor(bytes memory code) internal pure returns (bytes memory) {
```

Comment:
This changes the returned address of 
https://github.com/OffchainLabs/token-bridge-contracts/blob/9907fa26bfdec30dbdf8585192ad4ae6103f97f2/contracts/tokenbridge/ethereum/L1AtomicTokenBridgeCreator.sol#L612-L618
we might want to have a way to redeploy the multicall contract even if the rest of the token bridge is already deployed using the old way